src/workflows/airqo_etl_utils/data_validator.py

```
@@ -277,8 +277,23 @@ def process_data_for_api(data: pd.DataFrame) -> list:
+        Otherwise, it updates the checksum in XCom and returns the transformed DataFrame.
+
+        Args:
+            devices (pd.DataFrame): A Pandas DataFrame containing the devices data.
+            task_instance: The Airflow task instance used to pull and push XCom values.
+
+        Returns:
+            pd.DataFrame: Transformed DataFrame if the devices data has changed since
+                        the last execution; otherwise, an empty DataFrame.
+        """
+        from .airqo_utils import (
+            AirQoDataUtils,
```

Ruff

`.airqo_utils.AirQoDataUtils` imported but unused

Remove unused import: `.airqo_utils.AirQoDataUtils`

(F401)